cmd/activity-monitor/main.go

```diff
@@ -120,11 +96,11 @@ func startMonitor(rpcCh *amqp.Channel, logger *blog.AuditLogger, stats statsd.St
 		cmd.FailOnError(err, "Could not subscribe to queue")
 	}
 
-	deliveryTimings := make(map[string]time.Time)
+	rpcMonitor := metrics.NewRPCMonitor(stats)
 
 	// Run forever.
 	for d := range deliveries {
-		go timeDelivery(d, stats, deliveryTimings)
+		go rpcMonitor.TimeDelivery(d)
```

Comment:
Okay with that crazy map locking code, I'm now of the opinion that no maps should be involved and we should be carrying the context we need around the calls to send in the clients and servers.